components/Image/image-responsive.tsx

```diff
@@ -0,0 +1,288 @@
+import { useRef, useState, useEffect } from 'react'
+import { useAppContext } from '../../context/app-provider'
+
+type ImageProps = {
+  imgSource: string //base64 converted string for image
+  sourceType: number //type of source we are passing, default is url, 1= base64, and 2 = key to image string in app context
+  accelRate: number //the rate of change in panning speed
+  maxPanRate: number //maximum Pan Speed
+  zoomRate: number //rate of zoom, used for dblClick
+  zoomMax: number //maximum zoom of image
+  fHeight: number //set Height of the wrapper frame
+  fWidth: number //set Width of the wrapper frame
+}
+
+export default function ImageResponsive(props: ImageProps) {
+  const { images } = useAppContext()
+  const imgRef = useRef<HTMLImageElement>(null)
+  const { imgSource, sourceType, accelRate, maxPanRate, zoomRate, zoomMax, fHeight, fWidth } = props
+
+  const [imgTop, setImgTop] = useState(0) //translation of image from top: 0 <= imgTop <= topLimit
+  const [imgLeft, setImgLeft] = useState(0) //translation of image from top: 0 <= imgLeft <= leftLimit
+  const [topLimit, setTopLimit] = useState(0) //limit of translation in y direction (top) for current image size
+  const [leftLimit, setLeftLimit] = useState(0) //limit of translation in y direction (left) for current image size
+  const [isScaling, setIsScaling] = useState(false) //flag to set 'scaling' state, which allows zoooming of image
+  const [isDragging, setIsDragging] = useState(false) //flag to set 'dragging' state, which allows panning of image
+  const [orgHeight, setOrgHeight] = useState<number>(0) //The original height of image before any scaling
+  const [orgWidth, setOrgWidth] = useState<number>(0) //The original width of image before any scaling
+  const [scHeight, setScHeight] = useState<number>(0) //The current scaled height of image before any scaling
+  const [scWidth, setScWidth] = useState<number>(0) //The current scaled width of image before any scaling
+  const [imgScale, setImgScale] = useState(1) //The current scale rate for image
+  const [touchDist, setTouchDist] = useState(0) //The last recorded touch distance - used for pinch testing
+  const [isLoaded, setIsLoaded] = useState(false) //image has/has not loaded flag
+  const [offsetTop, setOffsetTop] = useState(0) //top offset from top most element. Required to accurate calculate mouse coordinates
+  const [offsetLeft, setOffsetLeft] = useState(0) //top offset from left most element. Required to accurate calculate mouse coordinates
+  const [oldXCoord, setOldXCoord] = useState(0) //stored previous mouse X coordinate
+  const [oldYCoord, setOldYCoord] = useState(0) //stored previous mouse Y coordinate
+  const [source, setSource] = useState<string>() //image source string as determined by passed parameter options
+  const [scaleInc, setScaleInc] = useState(0.025) //increment of scaling allowed
+  const [panRate, setPanRate] = useState(1) //current panning speed
```

Comment:
There's a lot of state based variables here. I wonder if it would make sense to create a config object and put these in there. That way you don't have to call setters everywhere and can have one setter for your config object. You'll just need to keep in mind that with objects in JS/TS you need to properly clone (deep copy) the object. Something else I wonder is if all the setter functions are being used. This is the most state based variables I've seen in a component and it's looking very busy.